configure.ac

```diff
@@ -2309,7 +2309,24 @@ fi
 
 AC_CHECK_FUNCS(__builtin_extract_return_address)
 
-AC_CHECK_HEADERS(unwind.h)
+# Enable unwind if found by default
+AC_ARG_WITH(unwind,
+  AS_HELP_STRING([--without-unwind], [Ignore unwind if found and disable it]))
+
+AS_IF([test "x$with_unwind" != "xno"],
+  [AC_CHECK_HEADERS(unwind.h)] [AC_CHECK_FUNCS(_Unwind_GetIP)],
+  [have_unwind=no])
+
+AS_IF([test "x$ac_cv_func__Unwind_GetIP" = "xyes"],
+  [have_unwind=yes],
```

Comment:
Here it looks like you want to use HAVE_UNWIND in the code but in NSException and config.h you use WITH_UNWIND. Overall the checks here look a bit confusing, could these be simplified.

Reply:
have_unwind is just a temporary variable, it does not get exported, WITH_UNWIND yes.

The test are simple: first check for the header, then if it exists, for an exported symbol and then check this against the original user choice.